src/compiler/program.ts

```diff
@@ -291,6 +286,12 @@ module ts {
             }
         }
 
+        function getCompilerOptionsDiagnostics(): Diagnostic[]{
+            let allDiagnostics: Diagnostic[] = [];
+            addRange(allDiagnostics, diagnostics.getGlobalDiagnostics());

```

Comment:
It also gets the checker global diagnostics like cannot find Array. I did not want to pass a flag to it, do adding the new function seemed more appealing.